### PR TITLE
Endpoint to Trigger Notifications for all resource in the notification queue

### PR DESCRIPTION
**Pull request by aravindmd**
- Endpoint to Trigger Notifications for all resource in the notification queue
- Added a endpoint on the testing controller

### CODE REVIEW

**Comment by aravindmd** [comment]
cc @jeyrschabu @emjburns

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/model/OnDemandNotifyResourceData.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.netflix.spinnaker.swabbie.model
+
+import com.fasterxml.jackson.annotation.JsonTypeName
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+data class OnDemandNotifyResourceData(
```
Comment (jeyrschabu) [inline swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/model/OnDemandNotifyResourceData.kt]: Let's keep one of these only. We don't need to create a new one.

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/model/OnDemandNotifyResourceData.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  val resourceId: String,
+  var projectedDeletionStamp: Long,
+  var markTs: Long?,
+  var resourceOwner: String = "swabbie@spinnaker.io",
+  val account: String,
+  val location: String
+)
+
+@JsonTypeName("testResource")
+data class NotifyResource(
```
Comment (jeyrschabu) [inline swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/model/OnDemandNotifyResourceData.kt]: Use the testResource for this

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/notifications/NotificationSender.kt [hunk 3]
```diff
@@ -220,7 +220,7 @@ class NotificationSender(
       .monitorValue(size())
   }
 
-  private data class NotificationResourceData(
```
Comment (jeyrschabu) [inline swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/notifications/NotificationSender.kt]: nit: I'd prefer not leaking this class out. Let's keep it private and build out what the notificationSender requires from the caller code. (Not needed for now but should refactor after testing)

File: swabbie-web/src/main/kotlin/com/netflix/spinnaker/swabbie/controllers/TestingController.kt [hunk 4]
```diff
@@ -86,6 +92,37 @@ class TestingController(
...
+    @PathVariable namespace: String,
+    @RequestBody resource: OnDemandNotifyResourceData
+  ) {
+    val workConfiguration = findWorkConfiguration(SwabbieNamespace.namespaceParser(namespace))
+    val markedResource = createMarkedResource(workConfiguration,
+      resource.resourceId,
+      resource.resourceOwner,
+      resource.projectedDeletionStamp,
+      resource.markTs)
+    val notificationResourceData = NotificationSender.NotificationResourceData(
```
Comment (jeyrschabu) [inline swabbie-web/src/main/kotlin/com/netflix/spinnaker/swabbie/controllers/TestingController.kt]: see comment on keeping this class private. (for later refactor)

File: swabbie-web/src/main/kotlin/com/netflix/spinnaker/swabbie/controllers/TestingController.kt [hunk 5]
```diff
@@ -94,4 +131,25 @@ class TestingController(
...
+
+  private fun createMarkedResource(
+    workConfiguration: WorkConfiguration,
+    id: String,
+    owner: String,
+    projectedDeletionStamp: Long,
+    markTs: Long?
+  ): MarkedResource {
+    return MarkedResource(
+      resource = NotifyResource(resourceId = id,
```
Comment (jeyrschabu) [inline swabbie-web/src/main/kotlin/com/netflix/spinnaker/swabbie/controllers/TestingController.kt]: Use testResource

**Comment by aravindmd** [comment]
removed `OnDemandNotifyResourceData` using 'OnDemandMarkData' instead . Also updated to use `testResource`

**Comment by aravindmd** [comment]
> Going to merge this as-is @aravindmd. I'd like a followup to address the remaining items

I did not want to change this interface for other endpoints , but since they are mostly for testing i guess i can make it required. I'll update that with the next change.

**Pull request merged**